packages/hotdog/hotdog.spec

```diff
@@ -10,67 +10,32 @@
 %global gitrev b85b75576adbbd7e133b54d71ebc11a28acf40db
 %global shortrev %(c=%{gitrev}; echo ${c:0:7})
 
-%global gosysrev 8c9f86f7a55f5af45a9080f6ed3ac915382d369c
-%global gosysrevshort %(c=%{gosysrev}; echo ${c:0:7})
-
-%global runtimespec 1.0.2
-
-%global goselinux 1.10.1
-
-%global libcap 1.2.63
-
 Name: %{_cross_os}hotdog
 Version: 1.0.1
 Release: 1%{?dist}
 Summary: Tool with OCI hooks to run the Log4j Hot Patch in containers
 License: Apache-2.0
 URL: https://github.com/bottlerocket-os/hotdog
-Source0: https://%{goimport}/archive/%{gorev}/%{gorepo}-%{shortrev}.tar.gz
-Source1: https://github.com/opencontainers/runtime-spec/archive/v%{runtimespec}/runtime-spec-%{runtimespec}.tar.gz
-Source2: https://github.com/golang/sys/archive/%{gosysrev}/sys-%{gosysrevshort}.tar.gz
-Source3: https://github.com/opencontainers/selinux/archive/refs/tags/v%{goselinux}.tar.gz#/go-selinux-v%{goselinux}.tar.gz
-Source4: https://git.kernel.org/pub/scm/libs/libcap/libcap.git/snapshot/libcap-cap/v%{libcap}.tar.gz#/libcap-v%{libcap}.tar.gz
-
+Source0: %{gorepo}-%{shortrev}-bundled.tar.gz
```

Comment:
I'd like to have two source files in this case:
* Source0 - pristine upstream archive
* Source1 - just the vendored modules

Reply:
> I'd like to have two source files in this case:

Hmmm, what is the benefit of having a separate untouched upstream archive alongside it's vendored modules?

In this PRs current state, the archive is downloaded, unpacked, modified via `go mod vendor` (which populates it's dependencies into a `vendor` directory) and then tar'd back up into a file named by `bundle-output-path`. This gives a simple, **single file** interface to then use in the build spec. It also emulates what a Go project _would_ look like if it had vendored it's modules into the source control.

But, I suppose with separate archives for the upstream source and bundled modules, we could have greater control over what makes it into end result. I.e., if we needed to patch an upstream's dependency at build time, we'd want to do that out of band from the actual upstream source code which would be painful if it's all already packaged into one archive.  

I guess this is really about which piece of the package building process should we combine the vendored modules and the upstream source: in the lookaside cahce's archive? After `buildsys` has downloaded the archive into a single tar file that gets passed to the spec? Within the spec itself during build time?

Seems like for the most control, we'd want the last option, at buildtime?

Reply:
> Hmmm, what is the benefit of having a separate untouched upstream archive alongside it's vendored modules?

It feels surprising to have `Cargo.toml` list one file with its hash, and have the spec file switch over to using a different file with an unknown hash without any warning. If you don't know about the magic that `buildsys` is doing, you might think the hash isn't actually being used, which would suggest a security problem.

By having the bundle archive just cover the delta - e.g. the vendor directory - you don't run the risk of something happening to one of the source files that isn't expected to change, which would then be difficult to track down.

> Seems like for the most control, we'd want the last option, at buildtime?

This seems right to me, even if it means a second `%setup` invocation to extract the second archive.

Reply:
> I'd like to have two source files in this case

👍🏼 Two archives are now used as `Source0` (the upstream pristine archive) and `Source1` which has the vendored dependencies